packages/mobile/src/utils/recipient.ts

```diff
@@ -230,22 +230,37 @@ export const filterRecipientFactory = (recipients: Recipient[], shouldSort?: boo
 
 export const buildRecentRecipients = (
   allRecipients: RecipientWithContact[],
-  recentPhoneNumbers: string[],
-  defaultDisplayName: string
+  recentKeys: string[],
+  defaultDisplayNameNumber: string,
+  defaultDisplayNameAddress: string
 ): Recipient[] =>
-  recentPhoneNumbers
-    .map((recentNumber) => {
+  recentKeys
+    .map((recentKey) => {
       const recipientsWithContacts: Recipient[] = allRecipients.filter(
-        (recipient) => recipient.e164PhoneNumber === recentNumber
+        (recipient) =>
+          recipient.e164PhoneNumber === recentKey ||
+          (recipient.address && recipient.address === recentKey)
       )
       if (recipientsWithContacts.length > 0) {
         return recipientsWithContacts
       }
+
+      if (isValidAddress(recentKey)) {
+        const recipientWithAddress: RecipientWithAddress = {
+          kind: RecipientKind.Address,
+          displayName: defaultDisplayNameAddress,
+          displayKey: recentKey.substring(0, 17) + '...',
```

Comment:
As I mentioned above, I'm finding displayKey confusing.
Let's leave the logic of how we want to display these recipients up to the components. No need for us to strip down the data and store that here.